GoogleDataTransportCCTSupport/Tests/Unit/GDTCCTNanopbHelpersTest.m

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <XCTest/XCTest.h>
+
+#import <nanopb/pb_decode.h>
+
+#import "GDTCCTEventGenerator.h"
+#import "GDTCCTNanopbHelpers.h"
+
+@interface GDTCCTNanopbHelpersTest : XCTestCase
+
+/** An event generator for testing. */
+@property(nonatomic) GDTCCTEventGenerator *generator;
+
+@end
+
+@implementation GDTCCTNanopbHelpersTest
+
+- (void)setUp {
+  self.generator = [[GDTCCTEventGenerator alloc] init];
+}
+
+- (void)tearDown {
+  [super tearDown];
+  [self.generator deleteGeneratedFilesFromDisk];
+}
+
+/** Tests that the event generator is generating consistent events. */
+- (void)testGeneratingFiveConsistentEvents {
+  NSArray<GDTStoredEvent *> *events1 = [self.generator generateTheFiveConsistentStoredEvents];
+  NSArray<GDTStoredEvent *> *events2 = [self.generator generateTheFiveConsistentStoredEvents];
+  XCTAssertEqual(events1.count, events2.count);
+  XCTAssertEqual(events1.count, 5);
+  for (int i = 0; i < events1.count; i++) {
+    GDTStoredEvent *storedEvent1 = events1[i];
+    GDTStoredEvent *storedEvent2 = events2[i];
+    XCTAssertEqualObjects(storedEvent1, storedEvent2);
+  }
+}
+
+/** Tests constructing a batched log request. */
+- (void)testConstructBatchedLogRequest {
+  NSBundle *testBundle = [NSBundle bundleForClass:[self class]];
+  NSArray *testData = @[
+    @"message-32347456.dat", @"message-35458880.dat", @"message-39882816.dat",
+    @"message-40043840.dat", @"message-40657984.dat"
+  ];
+  NSMutableSet *storedEvents = [[NSMutableSet alloc] init];
+  for (NSString *dataFile in testData) {
+    NSURL *fileURL = [testBundle URLForResource:dataFile withExtension:nil];
+    XCTAssertNotNil(fileURL);
+    [storedEvents addObject:[_generator generateStoredEvent:GDTEventQosDefault fileURL:fileURL]];
+  }
+  gdt_cct_BatchedLogRequest batch;
+  XCTAssertNoThrow((batch = GDTCCTConstructBatchedLogRequest(@{@"1018" : storedEvents})));
```

Comment:
This is a test, so you may not care, but the batch message should have pb_release() called on it or else you'll leak memory.

Reply:
Missed that one!